### ISSUE TITLE
Fix block order in Block Inserter so they appear alphabetically

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#27904]
**Describe the bug**
It's incredibly difficult to find blocks in the Block Inserter because none of the blocks appear in alphabetical order. The worst offender is the Embed blocks. I have no idea what order they are actually showing in at the moment. They're certainly not showing in "Most used" order (and if that is supposed to be the ordering, then it doesn't work properly). 

When people are adding their content, they know what type of content they're adding, so listing the  blocks alphabetically makes sense. e.g. I want to add a PullQuote or I want to add an image. By listing the Blocks alphabetically within each section, it makes it easier to find them as they're not jumping around all the time (assuming the current order is supposed to be "most used"). I shouldn't have to search for a block every single time I want to add one, which is what I have to do most times now, simply because I can't easily find anything within the list of blocks.

I feel confident in saying that rarely select/insert a Paragraph block from the Block Inserter. Gutenberg automatically inserts an empty paragraph after any block that isn't a Paragraph, and if the last block is a Paragraph block, then you just hit "Enter" to insert an empty Paragraph Block. Also, if you're inserting a block above/below an existing block, then Gutenberg automatically inserts an empty Paragraph block, which you then need to convert to another block if you want something other than a Paragraph block. Therefore, showing the Paragraph block as the first block in the Text Blocks section, makes no sense.

<img width="1440" alt="screenshot_109" src="https://user-images.githubusercontent.com/2078245/103166120-1672e700-4873-11eb-8c39-1983bbca0b7c.png">

I've never added a Twitter, WordPress, SoundCloud, Spotify, or Flickr embed block, but I have used YouTube and Vimeo embeds, so I have no idea why they show in this order

**To reproduce**
1. Create new page
2. See how long it takes you to find the _Amazon Kindle_ Embed block in the Block Inserter (without searching)

**Expected behaviour**
Blocks should be listed alphabetically to make them easier to find and so they constantly appear in a predicable order.

**Editor version (please complete the following information):**
- WordPress version: 5.6
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default
- Theme: TwentyNineteen

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.0.1 (20B29)
 - Browser: Firefox 84.0.1

**Comment by skorasaurus** [comment]
For what it's worth, the block inserter uses a combination of frequency and recency and your issue of all embed blocks displaying first is mentioned in https://github.com/WordPress/gutenberg/issues/28573

**Comment by maddisondesigns** [comment]
@skorasaurus Listing the blocks Alphabetically within each section would resolve this issue and #28573

**Comment by ntsekouras** [comment]
> Listing the blocks Alphabetically within each section would resolve this issue and #28573

#28573 has nothing to do with the alphabetical order.

**Comment by maddisondesigns** [comment]
@ntsekouras If the blocks were alphabetical, then surely you don't need to worry about 'fixing' the proper frecency calculation, making #28573 no longer necessary.

**Comment by ntsekouras** [comment]
Even if this approach is chosen eventually (there is discussion going on at the attached PR), #28573 will still be useful for `Most used blocks`.

**Comment by jameelmoses** [comment]
would also like to see this, or at least a filter to be able to adjust ourselves

**Comment by t-hamano** [comment]
Is what's suggested here still valid? Technically, if we wanted to alphabetize just the embed block variations, we could just swap the order of the elements in this array.

**Comment by maddisondesigns** [comment]
@t-hamano Yes, this is still completely valid. The only ordering that makes sense and helps make it easier for users to find blocks, is by having them all ordered Alphabetically. Any other ordering, such as `Frequency`, suggested in #28573, presents the blocks in a jumbled mess. Having everything ordered Alphabetically ensures that they're shown in a consistent order, every single time! Consistency is one of the fundamental principles of good usability.

**Comment by t-hamano** [comment]
@maddisondesigns Thanks for the reply.

Are you proposing to sort all blocks alphabetically, not just Embed blocks? If so, I'm concerned that blocks that will be used frequently (Paragraph, Image, Buttons, etc.) will become less prominent.

That seems to be one of the reasons why #27914 was closed.

**Comment by maddisondesigns** [comment]
@t-hamano Yes, each section within the Block Inserter, should have their blocks sorted Alphabetically. i.e. all the. `Text` Blocks sorted alphabetically, all the `Media` blocks sorted alphabetically, etc. etc.

As I've mentioned, having them sorted alphabetically makes them easier to find. If I want to insert an Image Block, I'll know exactly where to find it, every single time. I don't need to go looking through a jumbled mess of Blocks to find the one that I actually want.

I mentioned this in my opening comment. Having the Paragraph block appear as the very first block (assuming it's the most used block), makes zero sense.  Users rarely select/insert a Paragraph block, because they don't need to. Gutenberg automatically inserts an empty paragraph after any block that isn't a Paragraph. If the last block is a Paragraph block, then you just hit "Enter" to insert a new empty Paragraph Block to continue writing. If you're inserting a block above or below an existing block, using the `Add Before` and `Add After` options from the Block Toolbar, then Gutenberg automatically inserts an empty Paragraph block for you as well. If you want a different block, then you'll need to convert that Paragraph block to some other block.

With all that said, sorting the blocks alphabetically is just commonsense. It's much easier find the Block that you actually want, even if it is the Paragraph Block, when all the blocks are shown in a consistent order, every single time you open the Block Inserter. I cant think of any other software that re-orders their menus items, based on how many times you've chosen a selected item. It's not done that way, because it's just incredibly poor usability. You shouldn't be re-arranging menus on people. That's what the Block Inserter essentially is, it's just a drop-down menu with a bunch of elements you can add to your page.
